src/api/MetadataQuery.js

```diff
@@ -0,0 +1,121 @@
+/**
+ * @flow
+ * @file Helper for the box metadata query api
+ * @author Box
+ */
+import { List } from 'immutable';
+
+import type { ColumnType, ConditionType, ConnectorType, SortOrderType } from '../features/query-bar/flowTypes';
+import Base from './Base';
+import { ERROR_CODE_METADATA_QUERY } from '../constants';
+
+class MetadataQuery extends Base {
+    /**
+     * URL for recents api
+     *
+     * @return {string} base url for files
+     */
+    getUrl(): string {
+        return `${this.getBaseApiUrl()}/metadata_queries/execute`;
+    }
+
+    /**
+     * Utility to build query post body from conditions
+     *
+     * @param {string} scope - metadata scope, usually global or enterprise_
+     * @param {string} templateKey - the name of the metadata template
+     * @param {List<ColumnType>} columns - the list of columns/fields in the template
+     * @param {List<ConditionType>} conditions - the list of query conditions
+     * @param {ColumnType} orderByColumn - the column to order by
+     * @param {SortOrderType} orderDirection - the typed sort order ('ASC', 'DESC')
+     * @param {ConnectorType} connector connector for query ('AND' / 'OR')
+     * @return {void}
+     */
+
+    createQueryPostBody = (
+        scope: string,
+        templateKey: string,
+        columns: List<ColumnType>,
+        conditions: List<ConditionType>,
+        orderByColumn: ColumnType,
+        orderDirection: SortOrderType,
+        connector: ConnectorType,
+    ) => {
+        const queryParams = {}; // Hashmap of arguments used in conditions
+        const conditionsList = []; // Array of conditions joined (by connector) to form the final query predicate
```

Comment:
It's hard to differentiate the purpose of `conditions` vs `conditionsList`. 

Alternative names might be `conditionStrings`, `stringifiedConditions`, `conditionsRightBeforeConnectorJoin`. Open to suggestions.